src/rpc/multipass.proto

```diff
@@ -159,14 +161,16 @@ message FindReply {
     string log_line = 5;
 }
 
-message InstanceNames {
-    repeated string instance_name = 1;
+message InstanceOrSnapshot {
+    string instance_name = 1;
+    optional string snapshot_name = 2; // if this is present, the msg specifies a snapshot; otherwise, an instance
```

Comment:
While we could reserve the empty string to mean "undefined", an `optional` type is usually better because it carries a dedicated and generic interface that avoids ambiguity.

Reply:
Ok, I see your point. Having the `optional` type also makes it more readable in that it communicates that the field does not need to be populated.